File: serene/src/Serene.Web/Modules/ServerTypes/Documents/DocumentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DocumentRow } from "./DocumentRow";
import { DocumentType } from "./DocumentType";

export interface DocumentColumns {
    Title: Column<DocumentRow>;
    DocumentType: Column<DocumentRow>;
    State: Column<DocumentRow>;
}

export class DocumentColumns extends ColumnsBase<DocumentRow> {
    static readonly columnsKey = 'Documents.Document';
    static readonly Fields = fieldsProxy<DocumentColumns>();
}

[DocumentType]; // referenced types

File: serene/src/Serene.Web/Modules/Documents/DocumentColumns.cs
namespace Serene.Documents.Forms;

[ColumnsScript("Documents.Document")]
[BasedOnRow(typeof(DocumentRow), CheckNames = true)]
public class DocumentColumns
{
    [EditLink]
    public string Title { get; set; }
    public DocumentType DocumentType { get; set; }
    public string State { get; set; }
}
